addons/path-mapped/common/src/main/java/org/commonjava/indy/pathmapped/inject/PathMappedGroupRepositoryFilter.java

```diff
@@ -81,7 +87,15 @@ public boolean canProcess( String path, Group group )
             return concreteStores;
         }
 
-        Set<String> ret = pathMappedFileManager.getFileSystemContainingDirectory( candidates, strategyPath );
+        // batch it to avoid huge 'IN' query
+        Set<String> ret = new HashSet<>();
```

Comment:
Seems this breaks one time whole operation into several times of sub operations which need to connect to cassandra. Not sure how many candidates in real env (3000+?). We can test if this will bring connection performance. (or maybe concurrent them?)

Reply:
The candidates are 3000+. Prod DA send such heavy calls on a concurrent level of tens or hundred+ threads. All those heavy calls break Cassandra (cause the Cassandra read timeout). This fix will slow it down. We will try different batch size to find a good match.